File: solutions/javascript/lasagna-master/1/lasagna-master.js
/// <reference path="./global.d.ts" />

// @ts-check

/**
 * Implement the functions needed to solve the exercise here.
 * Do not forget to export them so they are available for the
 * tests. Here an example of the syntax as reminder:
 * 
 * export function yourFunction(...) {
 *   ...
 * }
 
 * @param {number} remainingTime
 */
export function cookingStatus(remainingTime) {
    if (remainingTime === 0) {
	return "Lasagna is done."
    } else if (remainingTime > 0) {
	return "Not done, please wait."
    } else if (!remainingTime) { // not provided
	return "You forgot to set the timer."
    }
}

/**
 * @param {string | any[]} layers
 * @param {number} averageLayerPreparationTime
 */
export function preparationTime(layers, averageLayerPreparationTime) {
    if (!averageLayerPreparationTime) {
	averageLayerPreparationTime = 2
    }
    return layers.length * averageLayerPreparationTime
}

/**
 * @param {any} layers
 */
export function quantities (layers) {
    let noodlesQuantity = 0
    let sauceLitres = 0
    for (let i = 0; i < layers.length; i++) {
	if (layers[i] === "noodles") {
	    noodlesQuantity += 50
	} else if (layers[i] === "sauce") {
	    sauceLitres += 0.2
	}
    }
    return {'noodles': noodlesQuantity,
	    'sauce': sauceLitres,
	   }
}

export function addSecretIngredient (ingredients1, ingredients2) {
    let secretIngredient = ingredients1[ingredients1.length - 1]
    ingredients2.push(secretIngredient)
}

function scaleRecipe (recipe, portions) {
    return {'noodles': recipe['noodles'] * (portions / 2),
	    'sauce': recipe['sauce'] * (portions / 2),
	    'mozzarella': recipe['mozzarella'] * (portions / 2),
	    'meat': recipe['meat'] * (portions / 2)
	   }
}

File: solutions/go/lasagna-master/1/lasagna_master.go
package lasagnamaster

// TODO: define the 'PreparationTime()' function
func PreparationTime (layers []string, perSliceTime int) int {
    if perSliceTime == 0 {
        perSliceTime = 2
    }
    return perSliceTime * len(layers)
} 
// TODO: define the 'Quantities()' function
func Quantities(layers []string) (noodles int, sauce float64) {
    for i := 0; i < len(layers); i++ {
        if layers[i] == "noodles" {
            noodles += 50
        } else if layers[i] == "sauce" {
            sauce += 0.2
        }
    }
    return
}
// TODO: define the 'AddSecretIngredient()' function
func AddSecretIngredient(ingredients_1 []string, ingredients_2 []string) {
    ingredients_2[len(ingredients_2)- 1] = ingredients_1[len(ingredients_1) - 1] 
}
// TODO: define the 'ScaleRecipe()' function
func ScaleRecipe(amounts []float64, nr_portions int) []float64 {
    newSlice := make([]float64, len(amounts))
    for i := 0; i < len(amounts); i++ {
        newSlice[i] = amounts[i] * (float64(nr_portions) / 2.0)
    }
    return newSlice
}
// Your first steps could be to read through the tasks, and create
// these functions with their correct parameter lists and return types.
// The function body only needs to contain `panic("")`.
//
// This will make the tests compile, but they will fail.
// You can then implement the function logic one by one and see
// an increasing number of tests passing as you implement more
// functionality.
